sklearn/base.py

```diff
@@ -523,3 +526,51 @@ def is_classifier(estimator):
 def is_regressor(estimator):
     """Returns True if the given estimator is (probably) a regressor."""
     return getattr(estimator, "_estimator_type", None) == "regressor"
+
+
+class _FrozenFit(object):
+    # We use a class as this allows isinstance check in clone
+    def __init__(self, obj):
+        self.obj = obj
+
+    def __call__(self, *args, **kwargs):
+        return self.obj
+
+
+def _frozen_fit_method(obj, method, X, *args, **kwargs):
+    return getattr(obj, method)(X)
+
+
+def freeze(estimator, copy=False):
+    """Copies estimator and freezes it
+
+    Frozen estimators:
+        * have ``fit(self, *args, **kwargs)`` merely return ``self``
+        * have ``fit_transform`` merely perform ``transform``
+        * have ``fit_predict`` merely perform ``predict``
+
+    Parameters
+    ----------
+    estimator : estimator
+    copy : bool
+
+    Returns
+    -------
+    frozen_estimator : estimator
+        A frozen copy of the input estimator, if ``copy``; otherwise, the
+        estimator is mutated to a frozen version of itself.
+
+    Notes
+    -----
+    Only works on estimators with ``__dict__``.
+    """
+    if copy:
+        estimator = deepcopy(estimator)
+    estimator.fit = _FrozenFit(estimator)
+    if hasattr(estimator, 'fit_transform'):
```

Comment:
Or did you consider them required API? The transformer and cluster base classes have those, but they are not really part of the API contract imho.